docs: add CONTRIBUTING.md file as contributing guide (#29)
monstar-lab-oss/admin-panel-template-reactjs: CONTRIBUTING.md

# Contribution guide

> ### Project tools

Reviewer: Heading levels should only increment by one level at a time

```suggestion
## Project tools
```

> - Github for code and issues management and

Reviewer: and? 😊 
```suggestion
- Github for code and issues management
```

### Branch rules:

1. Following branches should be used for primary code management

   1. `master` there can be only this branch. This hold all the latest already released code.

   - in case we are doing tag based release, `master` can be used as stable bleeding edge releasable code.

2. `{type}/{GithubIssueNo}-issue-one-liner` should be the format for branch naming

>    1. see [Type](#__type) section for branch `{type}`s.

Reviewer: ```suggestion
   1. See [Type](#__type) section for branch `{type}`.
```

>    2. Find `{GitgubIssueNo}` in Github.

Reviewer: ```suggestion
   2. Find `{GithubIssueNo}` in Github.
```

### Pull requests

Pull requests are the only way to propose a value you want to add. Following is a general workflow for submitting any requests.

1. Clone the repo and create your branch from `master`.
2. If you've added code that should be tested

> 3. Ensure that your code don't failed to build

Reviewer: ```suggestion
3. Ensure that your code doesn't fail to build
```

4. Make pull request to `master` branch

> 5. Assign respectives to merge pull request

Reviewer: ```suggestion
5. Assign respective develops to review and merge pull request
```

### Commit Message Guidelines

We have very precise rules over how our git commit messages can be formatted. This leads to **more
readable messages** that are easy to follow when looking through the **project history**. But also,

> we use the git commit messages to **generate the change log**.

Reviewer: ```suggestion
we use the git commit messages to **generate the changelog**.
```
